src/os/filestore/FileJournal.cc

```diff
@@ -1217,9 +1217,11 @@ void FileJournal::write_thread_entry()
       // but should be fine given that we will have plenty of aios in
       // flight if we hit this limit to ensure we keep the device
       // saturated.
+      uint64_t min_size = 64 * 1024;
       while (aio_num > 0) {
 	int exp = MIN(aio_num * 2, 24);
 	long unsigned min_new = 1ull << exp;
+	min_new = min_new < min_size ? min_size : min_new;
```

Comment:
I would suggest a minimal fix.  The original approach looks wrong to me: it will make it harder to submit the second io, not easier.  And I'm hesitant to flip things around like the second suggestion does.

How about just
<pre>
if (cur >= min_new || aio_num < 2)
  break;
</pre>
or something such that we can always submit at least 2 aios?  Would that resolve the case you're seeing?